mango-core/src/main/scala/org/bdgenomics/mango/io/VcfReader.scala

```diff
@@ -0,0 +1,249 @@
+  }
+
+  /**
+   * Loads data from bam files (indexed or unindexed) from s3.
+   *
+   * @param regionsOpt Iterable of ReferenceRegions to load
+   * @param path filepath to load from
+   * @return Alignment dataset from the file over specified ReferenceRegion
+   */
+  def loadS3(path: String, regionsOpt: Option[Iterable[ReferenceRegion]]): Tuple2[VCFHeader, Array[ADAMVariantContext]] = {
+    throw new Exception("Not implemented")
+  }
+
+  /**
+   * Loads data from bam files (indexed or unindexed) from HDFS.
+   * @param sc SparkContext
+   * @param regionsOpt Option of iterable of ReferenceRegions to load
+   * @param fp filepath to load from
+   * @return Alignment dataset from the file over specified ReferenceRegion
+   */
+  def loadHDFS(sc: SparkContext, fp: String, regionsOpt: Option[Iterable[ReferenceRegion]]): Tuple2[VariantContextDataset, Array[ADAMVariantContext]] = {
+    val dataset =
+      if (regionsOpt.isDefined) {
+        val regions = regionsOpt.get
+        val predicateRegions: Iterable[ReferenceRegion] = regions
+          .flatMap(r => LazyMaterialization.getReferencePredicate(r))
+        sc.loadIndexedVcf(fp, predicateRegions)
+      } else {
+        sc.loadVcf(fp) // No regions specified, return all regions
+      }
+
+    (dataset, dataset.rdd.collect)
+  }
+
+  /**
+   * Generates an index for a VCF file.
+   * @param fp path to VCF or compressed VCF file
+   * @param codec VCFCodec
+   * @return path to index file
+   */
+  private def createIndex(fp: String, codec: VCFCodec): String = {
+
+    val file = new java.io.File(fp)
+    val isGzipped = fp.endsWith(".gz")
+
+    val idxFile: java.io.File = if (isGzipped) new java.io.File(file + ".tbi") else new java.io.File(file + ".idx")
+
+    // do not re-generate index file
+    if (!idxFile.exists()) {
+      log.warn(s"No index file for ${file.getAbsolutePath} found. Generating ${idxFile.getAbsolutePath}...")
+
+      // Create the index
+      if (!isGzipped) {
+        val idx = IndexFactory.createIntervalIndex(file, codec)
+
+        var stream: LittleEndianOutputStream = null
+        try {
+          stream = new LittleEndianOutputStream(new BufferedOutputStream(new FileOutputStream(idxFile)))
+          idx.write(stream)
+        } finally {
+          if (stream != null) {
+            stream.close()
+          }
+        }
+        idxFile.deleteOnExit()
+
+      } else {
+
+        // get sequences from header for index file
+        val tmpReader = new VCFFileReader(file, false)
+        val sequences = tmpReader.getFileHeader().getSequenceDictionary()
+        tmpReader.close()
+
+        val idx = IndexFactory.createTabixIndex(file, new VCFCodec(),
+          TabixFormat.VCF, sequences)
+
+        idx.write(idxFile)
+
+        idxFile.deleteOnExit()
+      }
+
+    }
+    idxFile.getAbsolutePath
+  }
+
+  // TODO already defined in ADAM in VariantContextConverter line 266
+  def getHeaderLines(header: VCFHeader): Seq[VCFHeaderLine] = {
```

Comment:
@heuermh this is also copied from ADAM, although it is pretty small so I don't think copying is the worst thing here